### PR TITLE
StandardizeIncrementFixer - fix handling static properties

### PR DESCRIPTION
**Pull request by kubawerlos**
Fixes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5042

### CODE REVIEW

File: tests/Fixer/Operator/StandardizeIncrementFixerTest.php [hunk 1]
```diff
@@ -526,6 +526,32 @@ public function provideFixCases()
...
+                        ++static::$v2;
+                    }
+                }',
+                '<?php class Foo {
+                    public static function bar() {
+                        self::$v1 += 1;
+                        static::$v2 += 1;
+                    }
+                }',
+            ],
```
Comment (guilliamxavier) [inline tests/Fixer/Operator/StandardizeIncrementFixerTest.php:554]: What about adding a test like this?

```php
            [
                '<?php ++$a::$b;',
                '<?php $a::$b += 1;',
            ],
```

Comment (kubawerlos) [reply tests/Fixer/Operator/StandardizeIncrementFixerTest.php:558]: Nice catch, added.

File: src/Fixer/Operator/StandardizeIncrementFixer.php [hunk 2]
```diff
@@ -135,8 +135,14 @@ private function findStart(Tokens $tokens, $index)
             $index = $tokens->getPrevMeaningfulToken($index);
         }
 
-        if ($tokens[$tokens->getPrevMeaningfulToken($index)]->isGivenKind(T_OBJECT_OPERATOR)) {
-            return $this->findStart($tokens, $tokens->getPrevMeaningfulToken($index));
+        $prevIndex = $tokens->getPrevMeaningfulToken($index);
+
+        if ($tokens[$prevIndex]->isGivenKind(T_DOUBLE_COLON)) {
+            return $tokens->getTokenNotOfKindSibling($prevIndex, -1, [[T_NS_SEPARATOR], [T_STATIC], [T_STRING], [T_VARIABLE]]) + 1;
```
Comment (guilliamxavier) [inline src/Fixer/Operator/StandardizeIncrementFixer.php:141]: Realizing that `$a->$b->$c += 1;` is handled recursively (below),
do we want to handle `$a::$b::$c += 1;` too? and mixed `$a->$b::$c += 1;` / `$a::$b->$c += 1;`? (maybe not...)

Comment (kubawerlos) [reply src/Fixer/Operator/StandardizeIncrementFixer.php:141]: Good catch, even too good, I think we need some more sophisticated check, see the test cases I've added.

File: tests/Fixer/Operator/StandardizeIncrementFixerTest.php [hunk 3]
```diff
@@ -526,6 +526,56 @@ public function provideFixCases()
...
+            [
+                '<?php ++$a->$b::$c->${$d}->${$e}::foo(1 + 2 * 3)->$g::$h;',
+                '<?php $a->$b::$c->${$d}->${$e}::foo(1 + 2 * 3)->$g::$h += 1;',
```
Comment (SpacePossum) [inline tests/Fixer/Operator/StandardizeIncrementFixerTest.php:577]: :see_no_evil:  :laughing:

File: src/Fixer/Operator/StandardizeIncrementFixer.php (hunk 2 above)
Comment (guilliamxavier) [reply src/Fixer/Operator/StandardizeIncrementFixer.php]: I see you managed to sort it out, good job! 👍

Comment (kubawerlos) [reply src/Fixer/Operator/StandardizeIncrementFixer.php]: @guilliamxavier did you take a look at the rest of PR to leave a review?

Comment (guilliamxavier) [reply src/Fixer/Operator/StandardizeIncrementFixer.php]: Yes I did look, I just didn't think to leave a review 😅

Comment (kubawerlos) [reply src/Fixer/Operator/StandardizeIncrementFixer.php]: Thanks

**Comment by SpacePossum** [comment]
Thank you @kubawerlos.

**Pull request merged**